### ISSUE TITLE
Maths validation doesn't seem to pull in parent namespace declarations

### ISSUE THREAD

**Issue by nickerso** [cellml/libcellml#175]
I'm getting validation errors such as:
```
Namespace prefix cellml for units on cn is not defined
```
in models where the `cellml` namespace prefix is declared on the `model` element. Might need to rework some existing validation tests which may not be expecting the correct errors.

**Comment by nickerso** [comment]
See also this comment in the `Parser::ParserImpl::loadComponent(...)` method:
```c++
// TODO: copy any namespaces declared in parents into the math element
//       so math is a valid subdocument.
```

**Comment by nickerso** [comment]
This was semi-addressed by #420, but further work to tidy up things is required. Closing this one in favour of #424.
